src/detectors/RPOTS/RomanPotsReconstruction_factory.cc

```diff
@@ -118,33 +121,35 @@ namespace eicrecon {
 
         for (const auto h: rawhits) {
 
-            // auto cellID = h->getCellID();
-            // The actual hit position in Global Coordinates
-            auto pos0 = h->getPosition();
+            auto cellID = h->getCellID();
+
+	    //global --> local begins here -----
+
+            auto gpos = converter->position(cellID);
 
-	    //TODO: this code is for the local coordinates conversion -- next PR
-            //auto gpos = converter->position(cellID);
-            // local positions
+	    // local positions
             //if (m_localDetElement.empty()) {
-            //    auto volman = detector->volumeManager();
-            //    local = volman.lookupDetElement(cellID);
-            //}
-            //auto pos0 = local.nominal().worldToLocal(
-            //        dd4hep::Position(gpos.x(), gpos.y(), gpos.z())); // hit position in local coordinates
+            auto volman = m_geoSvc->detector()->volumeManager();
+            local = volman.lookupDetElement(cellID);
+	    //}
+
+            auto pos0 = local.nominal().worldToLocal(dd4hep::Position(gpos.x(), gpos.y(), gpos.z())); // hit position in local coordinates
+
+	    //information is stored in cm, we need mm - multiply everything by 10 for now
 
-	    if(!goodHit2 && pos0.z > 27099.0 && pos0.z < 28022.0){
+	    if(!goodHit2 && 10*gpos.z() > 27099.0 && 10*gpos.z() < 28022.0){
 
-		goodHitX[1] = pos0.x;
-		goodHitY[1] = pos0.y;
-		goodHitZ[1] = pos0.z;
+		goodHitX[1] = 10*pos0.x();
```

Comment:
If 10 relates to unit conversions, please use dd4hep units instead of raw 10 which is really difficult to interpret

Reply:
Fixed.